File: sales-by-match/sales-by-match.js
//HackerRank - Sales by Match

function sockMerchant(n, ar) {
    // Write your code here
    if (n === 1) {
        return 0
    };
    const classify = new Map();
    let pairCount = 0;
    for (let i = 0; i < n; i++) {
        if (classify.has(ar[i])) {
            classify.set(ar[i], classify.get(ar[i]) + 1) 
        } else {
            classify.set(ar[i], 1)
        };
        if (classify.get(ar[i]) % 2 === 0) {
            pairCount += 1
        }
    };
    return pairCount
}

File: sales-by-match/sales-by-match.py
#hackerRank - Sales by Match

def sockMerchant(n, ar):
    # Write your code here
    pair_count = 0
    classify = {}
    for color in ar:
        if color in classify:
            classify[color] = classify[color] + 1
        else:
            classify[color] = 1
        
        if classify[color] % 2 == 0:
            pair_count += 1
    return pair_count